### PR TITLE
Submission for Onboarding Challenge

### PR DESCRIPTION
**Pull request by mangoqwq**

### CODE REVIEW

File: onboarding/source/amb_light_service.c [hunk 1]
```diff
@@ -18,22 +29,113 @@ static void lightServiceTask(void * pvParameters);
...
+
+    // Check if queue was created successfully
+    if (lightServiceQueue == NULL) {
+        return 0;
+    }
     /* USER CODE END */
     return 1;
 }
 
+static void adcStartConversion_selChn(adcBASE_t *adc, unsigned channel, unsigned fifo_size, unsigned group) {
```
Comment (dgobalak) [inline onboarding/source/amb_light_service.c]: To improve clarity, change all `unsigned` arguments/variables to the appropriate fixed-size integer type (like `uint8_t` or `uint32_t`)

File: onboarding/source/amb_light_service.c [hunk 2]
```diff
@@ -18,22 +29,113 @@ static void lightServiceTask(void * pvParameters);
...
+        sciPrintText(scilinREG, errorMsg, strlen((const char*) errorMsg));
+    }
     /* USER CODE END */
 }
 
 uint8_t sendToLightServiceQueue(light_event_t *event) {
     /* USER CODE BEGIN */
     // Send the event to the queue.
-    
+    if (event == MEASURE_LIGHT) {
```
Comment (dgobalak) [inline onboarding/source/amb_light_service.c]: Although this works for the challenge, it'd make more sense to send any event to the queue, not just MEASURE_LIGHT. I.e. no need for this if statement.

File: onboarding/source/amb_light_service.c [hunk 3]
```diff
@@ -18,22 +29,113 @@ static void lightServiceTask(void * pvParameters);
...
+    ASSERT(lightServiceTaskHandle != NULL);
+    ASSERT(lightServiceQueue != NULL);
+
+    light_event_t *light_event;
+    BaseType_t returned = xQueueReceive(lightServiceQueue,
+                                        &light_event,
+                                        (TickType_t) 0);
 
+    if (returned == pdPASS) {
+        uint32 sensor_data = getLightSensorData();
```
Comment (dgobalak) [inline onboarding/source/amb_light_service.c]: We typically use the fixed-size integer types with the `_t` suffix, such as `uint32_t`. They're both exactly the same, but it's good to keep things consistent.

Comment (dgobalak) [reply onboarding/source/amb_light_service.c]: Make this change anywhere you use `uint32`. You'll need to make sure that `stdint.h` is included

**Review by dgobalak — CHANGES_REQUESTED** [review]
Looks really good overall! Just a few minor comments.

**Comment by mangoqwq** [comment]
I fixed the requested changes!

File: onboarding/source/amb_light_service.c [hunk 4]
```diff
@@ -18,21 +29,110 @@ static void lightServiceTask(void * pvParameters);
...
+        unsigned char* errorMsg = (unsigned char*) "Error receiving light event from queue.\n";
+        sciPrintText(scilinREG, errorMsg, strlen((const char*) errorMsg));
+    }
     /* USER CODE END */
 }
 
 uint8_t sendToLightServiceQueue(light_event_t *event) {
     /* USER CODE BEGIN */
     // Send the event to the queue.
+    BaseType_t returned = xQueueSend(lightServiceQueue, (void *) &event, (TickType_t) 0);
```
Comment (dgobalak) [inline onboarding/source/amb_light_service.c]: `xQueueSend` expects a pointer to the event and `sendToLightServiceQueue` takes this pointer as an argument. So you should be passing `event` into xQueueSend, not `&event`

File: onboarding/source/controller.c [hunk 5]
```diff
@@ -94,7 +131,11 @@ static void ledTimerCallback(TimerHandle_t xTimer) {
 static void lightTimerCallback(TimerHandle_t xTimer) {
     /* USER CODE BEGIN */
     // Send light event to light service queue
-
+    uint8_t returned = sendToLightServiceQueue(MEASURE_LIGHT);
```
Comment (dgobalak) [inline onboarding/source/controller.c]: `sendToLightServiceQueue` expects a pointer to the event. So something like:
```
light_event_t event = MEASURE_LIGHT;
sendToLightServiceQueue(&event);
```

**Review by dgobalak — CHANGES_REQUESTED** [review]

**Comment by mangoqwq** [comment]
Fixed (hopefully).

**Comment by dgobalak** [comment]
Great job! If you want, you can test your program on the RM46 LaunchPad at the next work session. If not, feel free to pick up a new task to work on from the task board. I'll add you to the Notion and GitHub.

**Pull request closed without merging**